File: src/main/resources/static/js/service/protypeService.js
//服务层
app.service('protypeService',function($http){

    //读取列表数据绑定到表单中
    this.findAll=function(){
        return $http.get('../protype/findAll.do');
    }
    //分页
    this.findPage=function(page,rows){
        return $http.get('../protype/findPage.do?page='+page+'&rows='+rows);
    }
    //查询实体
    this.findOne=function(id){
        return $http.get('../protype/findOne.do?id='+id);
    }
    //增加
    this.add=function(entity){
        return  $http.post('../protype/add.do',entity );
    }
    //修改
    this.update=function(entity){
        return  $http.post('../protype/update.do',entity );
    }
    //删除
    this.dele=function(ids){
        return $http.get('../protype/delete.do?ids='+ids);
    }
    //搜索
    this.search=function(page,rows,searchEntity){
        return $http.post('../protype/search.do?page='+page+"&rows="+rows, searchEntity);
    }

    this.findByParentId = function(parentId){
        return $http.get("../protype/findByParentId.do?parentId="+parentId);
    }
});

File: src/main/java/com/zzti/fruits/service/ProTypeService.java
package com.zzti.fruits.service;


import com.zzti.fruits.pojo.Protype;

import java.util.List;

/*=========================================================================
Copyright (C), 2015-2019, 中原工学院
description: 商品分类服务接口
author: wangbaoqin
Date:2019/4/7  19:18
updateAuthor:  wangbaoqin
updateDescription: 
updateDate: 2019/4/19 
encoding: UTF-8
version: 1.0
=========================================================================*/
public interface ProTypeService {
    /**
     * 分页
     * @param parentId 上级id
     * @return
     */
    public List<Protype> findPage(String parentId);

    /**
     * 添加商品类型
     * @param protype
     */
    public  void add(Protype protype);

    /**
     * 查询单个商品分类
     * @param id
     * @return
     */
    Protype findOne(String id);

    /**
     * 修改商品分类
     * @param protype
     */
    void update(Protype protype);

    /**
     * 批量删除
     * @param ids
     */
    void delete(String[] ids);
}
